bigtable/beam/change-streams/src/test/java/SongRankTest.java

```diff
@@ -0,0 +1,108 @@
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+
+public class SongRankTest {
+
+  // This table needs to be created manually before running the test since there is no API to create
+  // change-stream enabled tables yet. For java-docs-samples, the table should already be created,
+  // but if deleted, run the create table command in the README.
+  private static final String TABLE_ID = "song-rank-test";
+  private static final String COLUMN_FAMILY_NAME = "cf";
+  private static final String COLUMN_NAME = "song";
+  private static final String TEST_OUTPUT_LOCATION = "test-output/";
+
+  private static String projectId;
+  private static String instanceId;
+  private ByteArrayOutputStream bout;
+
+  private static String requireEnv(String varName) {
+    String value = System.getenv(varName);
+    assertNotNull(
+        String.format("Environment variable '%s' is required to perform these tests.", varName),
+        value);
+    return value;
+  }
+
+  @BeforeClass
+  public static void beforeClass() {
+    projectId = requireEnv("GOOGLE_CLOUD_PROJECT");
+    instanceId = requireEnv("BIGTABLE_TESTING_INSTANCE");
+  }
+
+  @Before
+  public void setupStream() {
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+  }
+
+  @Test
+  public void testSongRank() throws IOException, InterruptedException {
+    String[] args = {"--bigtableProjectId=" + projectId, "--bigtableInstanceId=" + instanceId,
+        "--bigtableTableId=" + TABLE_ID, "--outputLocation=" + TEST_OUTPUT_LOCATION};
+
+    new Thread(() -> SongRank.main(args)).start();
+
+    // Pause for job to start.
+    Thread.sleep(10 * 1000);
+
+    BigtableDataClient dataClient = BigtableDataClient.create(projectId, instanceId);
+    String rowKey = "user-1234";
+    String song1 = "song 1-" + UUID.randomUUID().toString().substring(0, 5);
+    String song2 = "song 2-" + UUID.randomUUID().toString().substring(0, 5);
+
+    for (int i = 0; i < 3; i++) {
+      dataClient.mutateRow(
+          RowMutation.create(TABLE_ID, rowKey).setCell(COLUMN_FAMILY_NAME, COLUMN_NAME, song1));
+    }
+    dataClient.mutateRow(
+        RowMutation.create(TABLE_ID, rowKey).setCell(COLUMN_FAMILY_NAME, COLUMN_NAME, song2));
+
+    // Wait for output to be written
+    Thread.sleep(3 * 60 * 1000);
+
+    String output = bout.toString();
+    assertThat(output).contains("KV{" + song1 + ", 3}");
+    assertThat(output).contains("KV{" + song2 + ", 1}");
+
+    FileInputStream fis = new FileInputStream(
+        TEST_OUTPUT_LOCATION + "/song-charts/GlobalWindow-pane-0-00000-of-00001.txt");
+    byte[] data = new byte[(int) fis.available()];
+    fis.read(data);
+    String content = new String(data, StandardCharsets.UTF_8);
+    assertThat(content).contains("KV{" + song1 + ", 3}");
+    assertThat(content).contains("KV{" + song2 + ", 1}");
+
+    FileUtils.deleteDirectory(new File(TEST_OUTPUT_LOCATION));
```

Comment:
(optional) deleting the directory isn't necessary. it'll get deleted on the machine we use for testing. keeping it allows the user to take a look at the content of this test run. i'll leave it up to you.

Reply:
if the test fails it doesn't get deleted locally, so the user can see it that way and is nice to have it clean up when it succeeds